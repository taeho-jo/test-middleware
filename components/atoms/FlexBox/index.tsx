import React from 'react';

// Styles
import { css } from '@emotion/react';

interface PropsType {
  children: React.ReactNode;
  justify?:
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'flex-start'
    | 'flex-end';
  align?: string;
  width?: string;
  height?: string;
  padding?: string;
}

const FlexBox = ({
  children,
  justify = 'center',
  align = 'center',
  width = '100%',
  height = 'auto',
  padding = '10px 30px',
}: PropsType) => {
  return (
    <div css={flexBox(width, height, justify, align, padding)}>{children}</div>
  );
};

export default FlexBox;

const flexBox = (width, height, justify, align, padding) => css`
  width: ${width};
  height: ${height};
  display: flex;
  justify-content: ${justify};
  align-items: ${align};
  padding: ${padding};
`;
