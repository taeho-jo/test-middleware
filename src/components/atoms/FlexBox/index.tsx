import React from 'react';

// Styles
import { css } from '@emotion/react';
import { JustifyType } from '../../../common/types/commonTypes';

interface PropsType {
  children?: React.ReactNode;
  display?: 'block' | 'flex' | 'inline' | 'inline-block';
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
  justify?: JustifyType;
  align?: string;
  direction?: 'row' | 'column';
  width?: string;
  height?: string;
  padding?: string;
  onClick?: () => void;
  xs?: number;
  sm?: number;
  md?: number;
  lg?: number;
  backgroundColor?: string;
  border?: string;
  radius?: string;
  shadow?: string;
  column?: string;
  row?: string;
  style?: any;
}

const calcWidthPercent = (span: any) => {
  if (!span) return;

  const width = (span / 12) * 100;
  return width;
};

const FlexBox = ({
  children,
  display = 'flex',
  justify = 'center',
  align = 'center',
  width = '100%',
  height = 'auto',
  padding = '0',
  direction = 'row',
  onClick,
  wrap = 'nowrap',
  border = 'none',
  xs,
  sm,
  md = 12,
  lg = 12,
  backgroundColor,
  radius,
  shadow,
  column,
  row,
  style,
}: PropsType) => {
  return (
    <div
      onClick={onClick}
      css={[
        flexBox(
          xs,
          sm,
          md,
          lg,
          backgroundColor,
          width,
          height,
          justify,
          align,
          padding,
          direction,
          display,
          border,
          wrap,
          radius,
          shadow,
          column,
          row,
        ),
        style,
      ]}
    >
      {children}
    </div>
  );
};

export default FlexBox;

const flexBox = (
  xs,
  sm,
  md,
  lg,
  backgroundColor,
  width,
  height,
  justify,
  align,
  padding,
  direction,
  display,
  border,
  wrap,
  radius,
  shadow,
  column,
  row,
) => css`
  height: ${height};
  justify-content: ${justify};
  align-items: ${align};
  padding: ${padding};
  flex-direction: ${direction};
  background: ${backgroundColor};
  display: ${display};
  border: ${border};
  flex-wrap: ${wrap};
  overflow: hidden;
  border-radius: ${radius};
  box-shadow: ${shadow};
  grid-column: ${column};
  grid-row: ${row};
  width: ${lg ? `${calcWidthPercent(lg)}%` : 'auto'};
  @media only screen and (max-width: 1024px) {
    width: ${md ? `${calcWidthPercent(md)}%` : 'auto'};
  }
`;
