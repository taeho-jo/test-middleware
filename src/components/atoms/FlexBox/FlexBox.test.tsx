import React from 'react';
import { render } from '@testing-library/react';
import FlexBox from './index';

describe('Flex Box', () => {
  it('flex width grid', () => {
    const utils = render(<FlexBox lg={12}>테스트</FlexBox>);
    expect(utils.container).toMatchSnapshot();
  });

  it('flex width grid', () => {
    const utils = render(<FlexBox lg={10}>테스트</FlexBox>);
    expect(utils.container).toMatchSnapshot();
  });
  it('flex width grid', () => {
    const utils = render(<FlexBox md={12}>테스트</FlexBox>);
    expect(utils.container).toMatchSnapshot();
  });
});
