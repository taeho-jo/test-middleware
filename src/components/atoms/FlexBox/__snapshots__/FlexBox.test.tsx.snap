// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Box flex width grid 1`] = `
<div>
  <div
    class="css-14oyxa0-flexBox"
  >
    테스트
  </div>
</div>
`;

exports[`Flex Box flex width grid 2`] = `
<div>
  <div
    class="css-127x5qc-flexBox"
  >
    테스트
  </div>
</div>
`;

exports[`Flex Box flex width grid 3`] = `
<div>
  <div
    class="css-14oyxa0-flexBox"
  >
    테스트
  </div>
</div>
`;
